### ISSUE TITLE
Slider 1.00-beta.20 issues and errors

### ISSUE THREAD

**Issue by mihaimiculescu** [hperrin/svelte-material-ui#98]
Use case allowing the user to alter `value` either by moving the slider or by entering it in the `input`:
```
<Field >
<Input type="text" bind:value={value} on:change={() => recalc1()} />
</Field>
<FormField >
<Slider bind:value={value} step={0.1} />
</FormField>
<script>
................
  function recalc1() {
  value = (Math.round(10*value)/10);
  console.log("recalc1 done"); //for testing purposes only
}
...............
</script>
```        
*Expected behavior:*
When dragging the slider or nudging it via arrow keys, the value displayed should have max one decimal place.
*Actual behavior:*
Generally it works, but when moving the slider I get values such as `60.300000000000004` instead of `60.3`, or `60.400000000000006` instead of `60.4`. But `60.2` and respectively `60.5` are always correctly displayed.
So I changed the Slider to 
```
<Slider bind:value={value} on:input={() => recalc1()} step={0.1} />
```
No luck. `recalc1` does not get executed when I move the slider.
But with 
```
<Slider bind:value={value} on:MDCSlider:input={() => recalc1()} step={0.1} />
```
it does. Still, those pesky loooong decimal parts are still there when I use the arrow keys to move the slider.
Long story short, I got rid of those only by using alongside `on:MDCSlider:change`, like so:
```
<Slider bind:value={value} on:MDCSlider:input={() => recalc1()} on:MDCSlider:change={() => recalc1()} step={0.1} />
```
*Errors*

- Every time I touch an arrow key:
```
Unable to preventDefault inside passive event listener invocation. foundation.js:327 
MDCSliderFoundation.handleKeydown_ @ foundation.js:327
MDCSliderFoundation._this.keydownHandler_ @ foundation.js:66
```
- Vertical arrow keys also move the handle of the slider, while simultaneously scrolling the page.

However, these errors don't affect the functionality of the Slider.

**Comment by MintyMods** [comment]
@mihaimiculescu 

The decimal places behavior is actually coming from the underlying MDC component. 

Viewing the [following example](https://material-components-web.appspot.com/slider.html) and setting the step to be 0.1 as per your comments produces the same results in terms of decimal precision. 

See the following documentation on the [slider component](https://material.io/develop/web/components/input-controls/sliders/) for more information. 

What is strange is that the opening [issue comment](https://github.com/material-components/material-components-web/issues/25) shows a requirement to `quantize` the result yet the implementation is using a [parseFloat ](https://github.com/material-components/material-components-web/blob/da9b957758b77174a8f7a120c669f6daa169ab2f/packages/mdc-slider/component.ts#L200) function.

Suggested algorithm from issue comment:-

```
function quantize(val, min, max, step) {
  const numSteps = Math.round(val / step);
  const quantizedVal = numSteps * step;
  return Math.max(min, Math.min(max, quantizedVal));
}
```

Implementation from source:-

 ```
 private parseFloat_(str: string|null, defaultValue: number) {
    const num = parseFloat(str!);  // tslint:disable-line:ban
    const isNumeric = typeof num === 'number' && isFinite(num);
    return isNumeric ? num : defaultValue;
  }
```


Opening an issue against MDC for clarification on the intended functionality might be beneficial here?

**Comment by mihaimiculescu** [comment]
@MintyMods   Thanks for the reply.
Actually, after having it fixed with my `recalc1` function, I'm perfectly happy with the behavior. No more long decimal parts.

However, IMHO, the errors as described în the 2nd part of my previous comment, plus the strange behavior when using the up and down arrow keys, should be addressed.

**Comment by hperrin** [comment]
This is fixed in the latest version, 3.1.0. (It was fixed in MDC 11, and SMUI 3.1 migrated to MDC 11.)
